cmd/handler/main.go

```diff
@@ -254,12 +160,145 @@ func mainHandler() int {
 	setupLog.Info("starting manager")
 	if err = mgr.Start(ctrl.SetupSignalHandler()); err != nil {
 		setupLog.Error(err, "problem running manager")
-		return 1
+		return generalExitStatus
 	}
 
 	return 0
 }
 
+// Handler runs as a daemonset and we want that each handler pod will cache/reconcile only resources that belong the node it runs on.
+func cacheResourcesOnNodes(ctrlOptions *ctrl.Options) {
+	nodeName := environment.NodeName()
+	metadataNameMatchingNodeNameSelector := fields.Set{"metadata.name": nodeName}.AsSelector()
+	nodeLabelMatchingNodeNameSelector := labels.Set{nmstateapi.EnactmentNodeLabel: nodeName}.AsSelector()
+	ctrlOptions.NewCache = cache.BuilderWithOptions(cache.Options{
+		SelectorsByObject: cache.SelectorsByObject{
+			&corev1.Node{}: {
+				Field: metadataNameMatchingNodeNameSelector,
+			},
+			&nmstatev1beta1.NodeNetworkState{}: {
+				Field: metadataNameMatchingNodeNameSelector,
+			},
+			&nmstatev1beta1.NodeNetworkConfigurationEnactment{}: {
+				Label: nodeLabelMatchingNodeNameSelector,
+			},
+		},
+	})
+}
+
+func setupHandlerControllers(mgr manager.Manager) error {
+	if err := (&controllers.NodeReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("Node"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create Node controller", "controller", "NMState")
+		return err
+	}
+
+	apiClient, err := client.New(mgr.GetConfig(), client.Options{Scheme: mgr.GetScheme(), Mapper: mgr.GetRESTMapper()})
+	if err != nil {
+		setupLog.Error(err, "failed creating non cached client")
+		return err
+	}
+
+	if err = (&controllers.NodeNetworkConfigurationPolicyReconciler{
+		Client:    mgr.GetClient(),
+		APIClient: apiClient,
+		Log:       ctrl.Log.WithName("controllers").WithName("NodeNetworkConfigurationPolicy"),
+		Scheme:    mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create NodeNetworkConfigurationPolicy controller", "controller", "NMState")
+		return err
+	}
+
+	if err = (&controllers.NodeNetworkConfigurationEnactmentReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("NodeNetworkConfigurationEnactment"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create NodeNetworkConfigurationEnactment controller", "controller", "NMState")
+		return err
+	}
+
+	return nil
+}
+
+func createHealthyFile() error {
+	// Handler runs with host networking so opening ports is problematic
```

Comment:
This comment at this scope doesn't make much sense, it should be before the function call.